plugins/network-elements/nuage-vsp/src/com/cloud/agent/api/guru/ImplementNetworkVspCommand.java

```diff
@@ -33,16 +34,23 @@
     String _networkName;
     String _networkCidr;
     String _networkGateway;
+    Long _networkAclId;
+    List<String> _dnsServers;
+    List<String> _gatewaySystemIds;
     String _networkUuid;
     boolean _isL3Network;
+    boolean _isVpc;
+    boolean _isSharedNetwork;
     String _vpcName;
     String _vpcUuid;
     boolean _defaultEgressPolicy;
-    Collection<String> _ipAddressRange;
+    Collection<String[]> _ipAddressRange;
+    String _domainTemplateName;
 
     public ImplementNetworkVspCommand(String networkDomainName, String networkDomainPath, String networkDomainUuid, String networkAccountName, String networkAccountUuid,
-            String networkName, String networkCidr, String networkGateway, String networkUuid, boolean isL3Network, String vpcName, String vpcUuid, boolean defaultEgressPolicy,
-            Collection<String> ipAddressRange) {
+            String networkName, String networkCidr, String networkGateway, Long networkAclId, List<String> dnsServers, List<String> gatewaySystemIds, String networkUuid,
+            boolean isL3Network, boolean isVpc, boolean isSharedNetwork, String vpcName, String vpcUuid, boolean defaultEgressPolicy, Collection<String[]> ipAddressRange,
+            String domainTemplateName) {
```

Comment:
The number of parameters is extremely large -- creating a brittle interface and difficult to comprehend client client.  Please refactor to the Builder pattern (i.e. declare the builder a ``public static final`` inner class to create instances of ``ImplementNetworkVspCommand`` and declare the constructor of ``ImplementNetworkVspCommand`` ``private``).